docs/encyclopedia/application-message-passing.mdx

```diff
@@ -26,57 +28,85 @@ keywords:
+From the above descriptions, it may sound like Updates can do all the things that Queries and Signals can do. But, unlike Signals, Updates must be synchronous, that is, they must wait for the worker running the Workflow to acknowledge the request.
+
+Use Signals instead of Updates when:
+
+- You want to send a fully asynchronous message.
+- Your client does not want to rely on the worker being available.
+- Your Workflow’s client cares more about the latency of sending the message than it does of the end-to-end operation.
+- When you need to [Signal-With-Start](signal-with-start) atomically start the Workflow and then Signal as the first operation. (since there is currently no Update-with-start operation.)
+
+Use Updates instead of Signals when:
+
+- You want to track the completion of the message from the client.
+- You need a result or an exception from your message without having to query subsequently.
+- You’d like to “validate” the Update before accepting it into the Workflow and its history.
+- Your client cares more about the latency of the end-to-end operation than you care about the latency of sending the message.
+
+### For read requests
 
-A Signal delivers data to a running Workflow Execution.
-It cannot return data to the caller; to do so, use a [Query](#queries) instead.
-The Workflow code that handles a Signal can mutate Workflow state.
-A Signal can be sent from a Temporal Client or a Workflow.
-When a Signal is sent, it is received by the Temporal Service and recorded as an Event to the Workflow Execution [Event History](/workflows#event-history).
-A successful response from the Temporal Service means that the Signal has been persisted and will be delivered at least once to the Workflow Execution.[^1]
-The next scheduled Workflow Task will contain the Signal Event.
+You normally want to do a Query:
 
-A Signal must include a destination (Namespace and Workflow Id) and name.
-It can include a list of arguments.
+- Queries are efficient–they never add entries to the [Workflow Event History](/workflows#event-history).
+- Queries can operate on completed Workflows.
 
-Signal handlers are Workflow functions that listen for Signals by the Signal name.
-Signals are delivered in the order they are received by the Temporal Service and written to History.
-If multiple deliveries of a Signal would be a problem for your Workflow, add idempotency logic to your Signal handler that checks for duplicates.
+However, because Queries cannot block, sometimes Updates are best.\
+When your goal is to do a read once the Workflow achieves a certain desired state, you have two options.
 
-[^1]: The Temporal Service usually deduplicates Signals, but does not guarantee deduplication: During shard migration, two Signal Events (and therefore two deliveries to the Workflow Execution) can be recorded for a single Signal because the deduping info is stored only in memory.
+- You could poll periodically with Queries until the workflow is ready.
```

Comment:
Noticed a few instances workflow/worker are not capitalized - can we capitalize them please when talking about a Temporal Workflow/Worker? Thanks!